services/property-svc/main.go

```diff
@@ -19,7 +24,38 @@ const ServiceName = "property-svc"
 // Version is set at compile time
 var Version string
 
+func replay(ctx context.Context, eventStore *esdb.Client, propertyPostgresProjection *property_postgres_projection.Projection) error {
+	replay := eventstoredb.NewReplay(eventStore, func(ctx context.Context, event hwes.Event) error {
+		log.Info().Dict("event", event.GetZerologDict()).Msg("replay event to projections")
+		if err, _ := propertyPostgresProjection.HandleEvent(ctx, event); err != nil {
+			return err
+		}
+		return nil
+	}, &[]string{fmt.Sprintf("%s-", aggregate.PropertyAggregateType)})
+
+	replay.BeforeReplayHook = func() (err error) {
+		log.Info().Msg("BeforeReplayHook")
+
+		// TOOD: Run in transaction. Only commit after a successful .Run()
```

Comment:
@FoseFx could you take a look at this? Via `db.Begin()` to start a new transaction, I always got a "conn busy" error. We are already using the recommended pgxpool.

Reply:
Looks like sql.Scan is handled just-in-time (to prevent huge buffers I assume), which means the connection is still active at the time we are issuing the first `TRUNCATE` command, thus attempting to re-use the still active connection (which we just go the current tableName from).

I split it up in two phases, first collecting all tableNames, then truncating them.